src/test/java/org/junitpioneer/jupiter/EnvironmentVariableUtilsTests.java

```diff
@@ -82,14 +82,16 @@ void shouldModifyEnvironmentVariableIfPermissionIsGiven() {
 		 */
 		@SuppressWarnings("deprecated")
 		private void executeWithSecurityManager(Runnable runnable) {
-			java.security.Policy.getPolicy().refresh();
-			SecurityManager original = System.getSecurityManager();
-			System.setSecurityManager(new SecurityManager());
-			try {
-				runnable.run();
-			}
-			finally {
-				System.setSecurityManager(original);
+			if (Runtime.version().feature() < 24) {
```

Comment:
This brings me more and more to the thought that this extension reaches eol?

Reply:
At least given its current implementation. I'd like to see a feature variant based on Mockito (maybe also for system properties), similar to what https://github.com/webcompere/system-stubs is doing.

Reply:
Oh and can we test this change?

Reply:
I added the https://github.com/junit-pioneer/junit-pioneer/labels/full-build label to the PR - this should run the experimental java build (which is on 24).

Reply:
> This brings me more and more to the thought that this extension reaches eol?

This change is related only to [https://openjdk.org/jeps/486](JEP 486: Permanently Disable the Security Manager). I'm not sure how that is relevant to the extension?

Reply:
I ended up completely removing SecurityManager related code.

Reply:
> > This brings me more and more to the thought that this extension reaches eol?
> 
> This change is related only to [JEP 486: Permanently Disable the Security Manager](https://openjdk.org/jeps/486). I'm not sure how that is relevant to the extension?

I always have in mind that the system environment extension only works with the Security Manager, but you are completely right that these are different things.